salt/modules/boto_rds.py

```diff
@@ -493,9 +494,16 @@ def delete(name, skip_final_snapshot=None, final_db_snapshot_identifier=None,
     try:
         conn.delete_db_instance(name, skip_final_snapshot,
                                 final_db_snapshot_identifier)
-        msg = 'Deleted RDS instance {0}.'.format(name)
-        log.info(msg)
-        return True
+        if not wait_for_deletion:
+            log.info('Deleted RDS instance {0}.'.format(name))
+            return True
+        while True:
```

Comment:
We can't have this loop structured like this. Any provisioning failure will loop indefinietely.